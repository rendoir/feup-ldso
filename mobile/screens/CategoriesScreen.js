import React from 'react';
import { StyleSheet } from 'react-native';
import { Font, AppLoading } from "expo";
import { Root, Container, Content, List, Text, Button, ListItem } from 'native-base';
import axios from 'axios';
import Category from '../components/Category';
import NewCustomHeader from '../components/NewCustomHeader';

export default class CategoriesScreen extends React.Component {
    state = {
        loading: true,
        categories: []
    };

    static navigationOptions = {
        header: null
    };

    componentWillUnmount() {
        this.isCancelled = true;
    }

    async componentDidMount() {
        await Font.loadAsync({
            Roboto: require("native-base/Fonts/Roboto.ttf"),
            Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
        });
        this.getCategoriesFromApi();
        !this.isCancelled && this.setState({ loading: false });
    }

    getCategoriesFromApi() {
        let self = this;
        axios.get('http://' + global.api + ':3030/categories')
            .then(function(response) {
                const categories = response.data;
                if (!this.isCancelled)
                    self.setState({ categories });
            })
            .catch(function(error) {
                console.log(error);
            });
    }

    render() {
        if (this.state.loading) {
            return (
                <Root>
                    <AppLoading />
                </Root>
            );
        }
        const { navigate } = this.props.navigation;

        const categories = this.state.categories.map((category, i) => (
            <Category data={category} key={i} onPress={() => { navigate('Agenda', { selectedCategory: category.name, selectedCategoryId: category.id }); this.props.navigation.state.params.onSelect({ updateCall: true }); } } />
        ));

        return (
            <Container>
                <NewCustomHeader text='Categorias' fave={false} />
                <Content>
                    <List>
                        <ListItem style={styles.listItem} onPress={() => { navigate('Agenda', { selectedCategory: 'Categoria', selectedCategoryId: 'null' }); this.props.navigation.state.params.onSelect({ updateCall: true }); } }>
                            <Button transparent onPress={() => { navigate('Agenda', { selectedCategory: 'Categoria', selectedCategoryId: 'null' }); this.props.navigation.state.params.onSelect({ updateCall: true }); }}><Text style={styles.buttonText}>Todos</Text></Button>
                        </ListItem>
                        {categories}
                    </List>
                </Content>
            </Container>
        );
    }
}

const styles = StyleSheet.create({
    listItem: {
        justifyContent: "center"
    },
    header: {
        backgroundColor: '#2c8f7f',
        height: 80
    },
    headerText: {
        color: 'white',
        fontSize: 30,
        textAlign: 'center',
        width: '100%',
        alignSelf: 'center'
    },
    buttonText: {
        color: 'black',
        fontFamily: 'OpenSans-Regular',
        fontSize: 18,
        paddingLeft: 5
    }
});
