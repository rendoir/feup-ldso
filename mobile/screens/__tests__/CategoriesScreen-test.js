import 'react-native';
import React from 'react';
import CategoriesScreen from '../CategoriesScreen';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders correctly', async () => {
    const tree = renderer.create(<CategoriesScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
