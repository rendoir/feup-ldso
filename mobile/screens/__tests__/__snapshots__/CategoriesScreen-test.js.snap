// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoriesScreen
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    screenProps={
      Object {
        "language": "PT",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <NewCustomHeader
          fave={false}
          language="PT"
          text="Categorias"
        />,
        <Styled(Content)>
          <Styled(List)>
            <Styled(ListItem)
              className="list-categories"
              onPress={[Function]}
              style={
                Object {
                  "justifyContent": "center",
                }
              }
            >
              <Styled(Button)
                className="all-categories"
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Text)
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 18,
                      "paddingLeft": 5,
                    }
                  }
                >
                  Todos
                </Styled(Text)>
              </Styled(Button)>
            </Styled(ListItem)>
          </Styled(List)>
        </Styled(Content)>,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fave": false,
          "language": "PT",
          "text": "Categorias",
          "toggleLanguage": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(List)>
            <Styled(ListItem)
              className="list-categories"
              onPress={[Function]}
              style={
                Object {
                  "justifyContent": "center",
                }
              }
            >
              <Styled(Button)
                className="all-categories"
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Text)
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 18,
                      "paddingLeft": 5,
                    }
                  }
                >
                  Todos
                </Styled(Text)>
              </Styled(Button)>
            </Styled(ListItem)>
          </Styled(List)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(ListItem)
                className="list-categories"
                onPress={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>
              </Styled(ListItem)>,
              Array [],
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>,
                "className": "list-categories",
                "onPress": [Function],
                "style": Object {
                  "justifyContent": "center",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>,
                  "className": "all-categories",
                  "onPress": [Function],
                  "transparent": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Todos",
                    "style": Object {
                      "color": "black",
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 18,
                      "paddingLeft": 5,
                    },
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Todos",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NewCustomHeader
            fave={false}
            language="PT"
            text="Categorias"
          />,
          <Styled(Content)>
            <Styled(List)>
              <Styled(ListItem)
                className="list-categories"
                onPress={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>
              </Styled(ListItem)>
            </Styled(List)>
          </Styled(Content)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fave": false,
            "language": "PT",
            "text": "Categorias",
            "toggleLanguage": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(List)>
              <Styled(ListItem)
                className="list-categories"
                onPress={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>
              </Styled(ListItem)>
            </Styled(List)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(ListItem)
                  className="list-categories"
                  onPress={[Function]}
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                >
                  <Styled(Button)
                    className="all-categories"
                    onPress={[Function]}
                    transparent={true}
                  >
                    <Styled(Text)
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 18,
                          "paddingLeft": 5,
                        }
                      }
                    >
                      Todos
                    </Styled(Text)>
                  </Styled(Button)>
                </Styled(ListItem)>,
                Array [],
              ],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Button)
                    className="all-categories"
                    onPress={[Function]}
                    transparent={true}
                  >
                    <Styled(Text)
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 18,
                          "paddingLeft": 5,
                        }
                      }
                    >
                      Todos
                    </Styled(Text)>
                  </Styled(Button)>,
                  "className": "list-categories",
                  "onPress": [Function],
                  "style": Object {
                    "justifyContent": "center",
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Text)
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 18,
                          "paddingLeft": 5,
                        }
                      }
                    >
                      Todos
                    </Styled(Text)>,
                    "className": "all-categories",
                    "onPress": [Function],
                    "transparent": true,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Todos",
                      "style": Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      },
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": "Todos",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App snapshot renders correctly with categories 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoriesScreen
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    screenProps={
      Object {
        "language": "PT",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <NewCustomHeader
          fave={false}
          language="PT"
          text="Categorias"
        />,
        <Styled(Content)>
          <Styled(List)>
            <Styled(ListItem)
              className="list-categories"
              onPress={[Function]}
              style={
                Object {
                  "justifyContent": "center",
                }
              }
            >
              <Styled(Button)
                className="all-categories"
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Text)
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 18,
                      "paddingLeft": 5,
                    }
                  }
                >
                  Todos
                </Styled(Text)>
              </Styled(Button)>
            </Styled(ListItem)>
            <Category
              data={
                Object {
                  "id": 1,
                  "name": "Category 1",
                }
              }
              language="PT"
              onPress={[Function]}
            />
            <Category
              data={
                Object {
                  "id": 2,
                  "name": "Category 2",
                }
              }
              language="PT"
              onPress={[Function]}
            />
            <Category
              data={
                Object {
                  "id": 3,
                  "name": "Category 3",
                }
              }
              language="PT"
              onPress={[Function]}
            />
          </Styled(List)>
        </Styled(Content)>,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fave": false,
          "language": "PT",
          "text": "Categorias",
          "toggleLanguage": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(List)>
            <Styled(ListItem)
              className="list-categories"
              onPress={[Function]}
              style={
                Object {
                  "justifyContent": "center",
                }
              }
            >
              <Styled(Button)
                className="all-categories"
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Text)
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 18,
                      "paddingLeft": 5,
                    }
                  }
                >
                  Todos
                </Styled(Text)>
              </Styled(Button)>
            </Styled(ListItem)>
            <Category
              data={
                Object {
                  "id": 1,
                  "name": "Category 1",
                }
              }
              language="PT"
              onPress={[Function]}
            />
            <Category
              data={
                Object {
                  "id": 2,
                  "name": "Category 2",
                }
              }
              language="PT"
              onPress={[Function]}
            />
            <Category
              data={
                Object {
                  "id": 3,
                  "name": "Category 3",
                }
              }
              language="PT"
              onPress={[Function]}
            />
          </Styled(List)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(ListItem)
                className="list-categories"
                onPress={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>
              </Styled(ListItem)>,
              Array [
                <Category
                  data={
                    Object {
                      "id": 1,
                      "name": "Category 1",
                    }
                  }
                  language="PT"
                  onPress={[Function]}
                />,
                <Category
                  data={
                    Object {
                      "id": 2,
                      "name": "Category 2",
                    }
                  }
                  language="PT"
                  onPress={[Function]}
                />,
                <Category
                  data={
                    Object {
                      "id": 3,
                      "name": "Category 3",
                    }
                  }
                  language="PT"
                  onPress={[Function]}
                />,
              ],
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>,
                "className": "list-categories",
                "onPress": [Function],
                "style": Object {
                  "justifyContent": "center",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>,
                  "className": "all-categories",
                  "onPress": [Function],
                  "transparent": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Todos",
                    "style": Object {
                      "color": "black",
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 18,
                      "paddingLeft": 5,
                    },
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Todos",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "data": Object {
                  "id": 1,
                  "name": "Category 1",
                },
                "language": "PT",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "data": Object {
                  "id": 2,
                  "name": "Category 2",
                },
                "language": "PT",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "data": Object {
                  "id": 3,
                  "name": "Category 3",
                },
                "language": "PT",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NewCustomHeader
            fave={false}
            language="PT"
            text="Categorias"
          />,
          <Styled(Content)>
            <Styled(List)>
              <Styled(ListItem)
                className="list-categories"
                onPress={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>
              </Styled(ListItem)>
              <Category
                data={
                  Object {
                    "id": 1,
                    "name": "Category 1",
                  }
                }
                language="PT"
                onPress={[Function]}
              />
              <Category
                data={
                  Object {
                    "id": 2,
                    "name": "Category 2",
                  }
                }
                language="PT"
                onPress={[Function]}
              />
              <Category
                data={
                  Object {
                    "id": 3,
                    "name": "Category 3",
                  }
                }
                language="PT"
                onPress={[Function]}
              />
            </Styled(List)>
          </Styled(Content)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fave": false,
            "language": "PT",
            "text": "Categorias",
            "toggleLanguage": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(List)>
              <Styled(ListItem)
                className="list-categories"
                onPress={[Function]}
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Styled(Button)
                  className="all-categories"
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Text)
                    style={
                      Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      }
                    }
                  >
                    Todos
                  </Styled(Text)>
                </Styled(Button)>
              </Styled(ListItem)>
              <Category
                data={
                  Object {
                    "id": 1,
                    "name": "Category 1",
                  }
                }
                language="PT"
                onPress={[Function]}
              />
              <Category
                data={
                  Object {
                    "id": 2,
                    "name": "Category 2",
                  }
                }
                language="PT"
                onPress={[Function]}
              />
              <Category
                data={
                  Object {
                    "id": 3,
                    "name": "Category 3",
                  }
                }
                language="PT"
                onPress={[Function]}
              />
            </Styled(List)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(ListItem)
                  className="list-categories"
                  onPress={[Function]}
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                >
                  <Styled(Button)
                    className="all-categories"
                    onPress={[Function]}
                    transparent={true}
                  >
                    <Styled(Text)
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 18,
                          "paddingLeft": 5,
                        }
                      }
                    >
                      Todos
                    </Styled(Text)>
                  </Styled(Button)>
                </Styled(ListItem)>,
                Array [
                  <Category
                    data={
                      Object {
                        "id": 1,
                        "name": "Category 1",
                      }
                    }
                    language="PT"
                    onPress={[Function]}
                  />,
                  <Category
                    data={
                      Object {
                        "id": 2,
                        "name": "Category 2",
                      }
                    }
                    language="PT"
                    onPress={[Function]}
                  />,
                  <Category
                    data={
                      Object {
                        "id": 3,
                        "name": "Category 3",
                      }
                    }
                    language="PT"
                    onPress={[Function]}
                  />,
                ],
              ],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Button)
                    className="all-categories"
                    onPress={[Function]}
                    transparent={true}
                  >
                    <Styled(Text)
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 18,
                          "paddingLeft": 5,
                        }
                      }
                    >
                      Todos
                    </Styled(Text)>
                  </Styled(Button)>,
                  "className": "list-categories",
                  "onPress": [Function],
                  "style": Object {
                    "justifyContent": "center",
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Styled(Text)
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 18,
                          "paddingLeft": 5,
                        }
                      }
                    >
                      Todos
                    </Styled(Text)>,
                    "className": "all-categories",
                    "onPress": [Function],
                    "transparent": true,
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Todos",
                      "style": Object {
                        "color": "black",
                        "fontFamily": "OpenSans-Regular",
                        "fontSize": 18,
                        "paddingLeft": 5,
                      },
                      "virtual": undefined,
                    },
                    "ref": null,
                    "rendered": "Todos",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "data": Object {
                    "id": 1,
                    "name": "Category 1",
                  },
                  "language": "PT",
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "data": Object {
                    "id": 2,
                    "name": "Category 2",
                  },
                  "language": "PT",
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "data": Object {
                    "id": 3,
                    "name": "Category 3",
                  },
                  "language": "PT",
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
